vector/src/main/java/im/vector/app/features/home/room/threads/ThreadsManager.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.threads
+
+import android.app.Activity
+import android.content.Context
+import android.text.Spanned
+import androidx.core.text.HtmlCompat
+import im.vector.app.R
+import im.vector.app.features.MainActivity
+import im.vector.app.features.MainActivityArgs
+import im.vector.app.features.settings.VectorPreferences
+import org.matrix.android.sdk.internal.database.lightweight.LightweightSettingsStorage
+import javax.inject.Inject
+
+/**
+ * The class is responsible for handling thread specific tasks
+ */
+class ThreadsManager @Inject constructor(
+        private val vectorPreferences: VectorPreferences,
+        private val lightweightSettingsStorage: LightweightSettingsStorage,
+        private val context: Context
+) {
+
+    companion object {
+        const val learnMoreUrl = "https://element.io/help#threads"
+    }
+
+    /**
+     * Enable threads and invoke an initial sync. The initial sync is mandatory in order to change
+     * the already saved DB schema for already received messages
+     */
+    fun enableThreadsAndRestart(activity: Activity) {
+        vectorPreferences.setThreadMessagesEnabled()
+        lightweightSettingsStorage.setThreadMessagesEnabled(vectorPreferences.areThreadMessagesEnabled())
+        MainActivity.restartApp(activity, MainActivityArgs(clearCache = true))
+    }
+
+    /**
+     * Generates and return an Html spanned string to be rendered especially in dialogs
+     */
+    fun getBetaEnableThreadsMessage(): Spanned {
+        val href = "<a href='$learnMoreUrl'>Learn more</a>.<br><br>"
```

Comment:
if we want to allow translations here we'll need to move this `Learn more` to the resources

Reply:
thinking a little more, assuming the `learnMoreUrl` doesn't need to be dynamic, it could be in the string resource (we've done something similar for the EMS copy in the onboarding)

Reply:
The easiest way to implement that functionality was to include `<a href='$learnMoreUrl'>Learn more</a>.<br><br>` in a string resources but I didn't want to do that. Thats why I decoupled the html part with the rest.

1. Yes I agree` learn more` can be also a string resource, will update it accordingly
2. I have not seen any `http` links in any of our string resources so I tried to avoid it. And if we do that we should mark it as not translatable to avoid trying to translate them. Where exactly we do that in the onboarding?

Reply:
We already have several instance of "learn more" in the string, maybe create a generic one, and we can do some cleanup during the next welbate sync.

Reply:
for `2.`, I've just realised it hasn't been pushed yet whilst we figure out the exact url :facepalm:  the benefit of including the url in the translatable strings is that we could also change the domain based on locale eg DE could use the .de domain myurl.de 

although this only works if the url is static :thinking:

Reply:
@bmarty I also observed it will do a clean up there.

@ouchadam I would suggest to create a separate string resource file for non translatable strings. And when we have dynamic URLs with multiple domains for sure we can add them in our translatable strings

Reply:
If we want to translate the URL, this is fine, but please let the formatting out of the string resource.
Also I think the URL can be translated, but out of Weblate, since this is quite technical.

Reply:
example of url in the resources: https://github.com/vector-im/element-android/blob/main/vector-config/src/main/res/values/config.xml#L7